kiwi/builder/disk.py

```diff
@@ -785,7 +787,8 @@ def _build_and_map_disk_partitions(self, disksize_mbytes):  # noqa: C901
             )
             disksize_used_mbytes += squashed_rootfs_mbsize
 
-        if self.spare_part_mbsize and self.spare_part_is_last:
+        if self.spare_part_mbsize and self.spare_part_is_last and not \
+           self.disk_resize_requested:
```

Comment:
> The consistency of spare partition options together with resizing options is validated on the runtime check. 

this is correct, but we also allow to disable runtime checks. Assuming the runtime check is disabled and the code above does not look at disk_resize_requested and someone configures spare_part_is_last together with oem-resize enabled it will lead to a system that will fail the repart step on boot. I think the failed repart will still lead the system to boot up. So the question is what is better:

a) disabled runtime check and invalid config leads to a system that fails the repart on boot
b) disabled runtime check and invalid config leads to a system that ignores spare_part_is_last

maybe option a) is better as it tries to setup the requested but fails and there is also an error message about it.
In this case I would just delete the self.disk_resize_requested check in the if conditions.

> Also wondering if the sanity check couldn't be done with some schematron rule.

That might be the best solution. I will look into it

Reply:
I have created a schematron rule which works for me. The only problem I see with that is that we can't display longer text information as it all prints it in one line. See the new commit: Thoughts ?